### PR TITLE
Port `renorm` cpu from TH to ATen

### LINKED ISSUE #24754: Migrate `renorm` and `renorm_` from the TH to Aten (CPU)
Porting TH operators is essential for code simplicity and performance reasons.

Porting guides and Q&A are available in umbrella issue: #24507

Feel free to add @VitalyFedyunin as a reviewer to get a prioritized review.

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser

### PR DESCRIPTION
**Pull request by shihongzhi**
Fixes: #24754

### CODE REVIEW

**Comment by vincentqb** [comment]
@VitalyFedyunin -- since you are on #24754, can you review this PR?

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -1465,4 +1466,76 @@ Tensor& diag_out(Tensor &result, const Tensor& self, int64_t dimension) {
...
+}
+
+Tensor& renorm_(Tensor& self, Scalar p, int64_t dim, Scalar maxnorm) {
+  at::renorm_out(self, self, p, dim, maxnorm);
+  return self;
+}
+
+Tensor& renorm_out(Tensor &result, const Tensor& self, Scalar p, int64_t dim, Scalar maxnorm) {
+  AT_DISPATCH_FLOATING_TYPES(self.scalar_type(), "renorm", [&] {
+    apply_renorm<scalar_t>(result.contiguous(), self.contiguous(), p.to<scalar_t>(), dim, maxnorm.to<scalar_t>());
```
Comment (VitalyFedyunin) [inline aten/src/ATen/native/TensorShape.cpp]: If you are making self and result contiguous, you will have problems with `x.renorm_`called on non-contiguous tensor. Please add such test and fix it.

**Review by VitalyFedyunin — CHANGES_REQUESTED** [review]
Problems with non-contiguous tensors. Also please add benchmarks comparing old and new implementations.

**Comment by nikitaved** [comment]
Hi, @shihongzhi , are you still working on this?

**Comment by glaringlee** [comment]
> Hi, @shihongzhi , are you still working on this?

@nikitaved, since this is no longer updated, we will use it as an onboarding task for a new member in the team.

**Pull request closed without merging**